### PR TITLE
ENG-607538 : New storage class along with old.

### PR DESCRIPTION
**Pull request by koushikmalga**
Created option to deploy new storage class and option for letting the old storage class available.

### CODE REVIEW

File: charts/nutanix-csi-storage/templates/ntnx-sc.yaml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {{- if eq .Values.volumeClass true }}
 
-{{- $storageContainer := .Values.storageContainer | required "storageContainer is required." -}}
+{{ $storageContainer := .Values.storageContainer | required "storageContainer is required." }}
 ---
```
Comment (subodh01) [inline charts/nutanix-csi-storage/templates/ntnx-sc.yaml:4]: We have to add secret parameters like it use to be in 2.6.x for nutanix-volumes
https://github.com/nutanix/helm/blob/master/charts/nutanix-csi-storage/templates/ntnx-sc.yaml#L23

File: charts/nutanix-csi-storage/values.yaml [hunk 2]
```diff
@@ -38,11 +38,19 @@ imagePullPolicy: IfNotPresent
 # Storage Class settings
 #
 # choose for which mode (Volume, File, Dynamic File) storageclass need to be created
+# If you created a snapshot using this Storage Class and upgrading from 2.6.x to 3.0.0
```
Comment (subodh01) [inline charts/nutanix-csi-storage/values.yaml]: If you have an existing VolumeSnapshot which was created using Nutanix CSI Driver < 3.0.0, you need a storageclass compatible with CSI Driver < 3.0.0 to restore a PV from that volume snapshot. Please set 
volumeClass to true for that.

**Review by subodh01 — CHANGES_REQUESTED** [review]

File: charts/nutanix-csi-storage/values.yaml [hunk 3]
```diff
@@ -55,7 +63,7 @@ dynamicFileClassRetention: "Delete"
 # Default Storage Class settings
 #
 # Decide wich storageclass will be the default
-# value are: node, volume, file, dynfile
+# value are: node, volume, volume-v3, file, dynfile
```
Comment (aniruddhg-nx) [inline charts/nutanix-csi-storage/values.yaml]: "values" or "possible values"

File: charts/nutanix-csi-storage/templates/ntnx-sc.yaml (hunk 1 above)
Comment (aniruddhg-nx) [reply charts/nutanix-csi-storage/templates/ntnx-sc.yaml:4]: we should also run some manual tests to verify helm changes. @koushikmalga

File: charts/nutanix-csi-storage/templates/ntnx-sc.yaml [hunk 4]
```diff
@@ -26,6 +32,39 @@ volumeBindingMode: WaitForFirstConsumer
...
+metadata:
+    name: {{ .Values.newVolumeClassName }}
+{{- if eq .Values.defaultStorageClass "volume-v3" }}
+    annotations:
+        storageclass.kubernetes.io/is-default-class: "true"
+{{- end }}
+provisioner: {{ include "nutanix-csi-storage.drivername" . }}
+parameters:
+    storageType: NutanixVolumes
+    csi.storage.k8s.io/provisioner-secret-name: {{ .Values.pcSecretName }}
```
Comment (subodh01) [inline charts/nutanix-csi-storage/templates/ntnx-sc.yaml]: Don't add secret for the new one.

**Review by subodh01 — APPROVED** [review]

**Pull request merged**